### PR TITLE
Update clear platform

### PR DESCRIPTION
**Pull request by novain1102**

### CODE REVIEW

File: native/utils/clear-platform.js [hunk 1]
```diff
@@ -2,43 +2,82 @@
...
-const { existsSync, removeSync } = require('fs-extra');
+const fs = require('fs-extra');
 
-// npm run clear-platform
+// usage: npm run clear-platform
 let externalDir = path.join(__dirname, "../external");
 let failed = false;
 
+let macUsefulDirs = ['android', 'cmake', 'emscripten', 'ios', 'ios-m1-simulator', 'mac', 'ohos', 'sources'];
+let winUsefulDirs = ['android', 'cmake', 'emscripten', 'ohos', 'sources', 'win32', 'win64'];
```
Comment (minggo) [inline native/utils/clear-platform.js]: win32 is removed in v3.6.0.

File: native/utils/clear-platform.js [hunk 2]
```diff
@@ -2,43 +2,82 @@
...
-        ensureRemove(clearDirectory);
-        clearDirectory = path.join(externalDir, "./ios");
-        console.log(`  ${chalk.green('Remove directory: ')} ${clearDirectory}`);
-        ensureRemove(clearDirectory);
-        clearDirectory = path.join(externalDir, "./ios-m1-simulator");
-        console.log(`  ${chalk.green('Remove directory: ')} ${clearDirectory}`);
-        ensureRemove(clearDirectory);
+function cleanPlatform(platform) {
+    if (platform === 'darwin') { // macOS
+        let macUselessDirs = getArrDiff(allDirs, macUsefulDirs);
```
Comment (minggo) [inline native/utils/clear-platform.js:39]: Why not just use `macUsefulDirs ` as `allDirs` is empty.

Comment (novain1102) [reply native/utils/clear-platform.js:39]: `allDirs` will get all dirs in external firstly in main function.

File: native/utils/clear-platform.js (hunk 1 above)
Comment (novain1102) [reply native/utils/clear-platform.js]: got it. already removed.

**Review by minggo — APPROVED** [review]

**Pull request merged**